packages/iframe-wallet-poc/src/iframe/user.ts

```diff
@@ -249,14 +249,14 @@ async function signMyUpdate(
   next: Function,
   context: machine.state.Context,
   user: User
-): Promise<cf.utils.Signature> {
+): Promise<ethers.utils.Signature> {
   const operation: machine.cfTypes.CfOperation = machine.middleware.getFirstResult(
     machine.instructions.Instruction.OP_GENERATE,
     context.results
   ).value;
   const digest = operation.hashToSign();
-  const sig = user.signingKey.signDigest(digest);
-  return new cf.utils.Signature(sig.recoveryParam! + 27, sig.r, sig.s);
+  const { recoveryParam, r, s } = user.signingKey.signDigest(digest);
+  return { r, s, v: recoveryParam! + 27 };
```

Comment:
Is returning this object equivalent to returning an instance of an `ethers.utils.Signature`?

Even if so, wouldn't be easier to parse if a signature instance is constructed using this return data and have the instance be returned?

Reply:
> Is returning this object equivalent to returning an instance of an ethers.utils.Signature?
Yup

There is no such thing as a "Signature instance", it's just an interface now.